app/main/views/users.py

```diff
@@ -121,6 +121,62 @@ def download_users(framework_slug):
     )
 
 
+@main.route('/frameworks/<framework_slug>/suppliers/download', methods=['GET'])
+@role_required('admin-framework-manager', 'admin')
+def download_suppliers(framework_slug):
+    framework = data_api_client.get_framework(framework_slug).get("frameworks")
+
+    supplier_rows = data_api_client.export_suppliers(framework_slug).get('suppliers', [])
+
+    supplier_and_framework_headers = [
+        "supplier_id",
+        "supplier_name",
+        "supplier_organisation_size",
+        "duns_number",
+        "companies_house_number",
+        "registered_name",
+        "declaration_status",
+        "application_status",
+        "application_result",
+        "framework_agreement",
+        "variations_agreed",
+    ]
+    service_count_headers = [lot['slug'] for lot in framework['lots']]
+    contact_info_headers = [
+        'contact_name',
+        'contact_email',
+        'contact_phone_number',
+        'address_first_line',
+        'address_city',
+        'address_postcode',
+        'address_country',
+    ]
+
+    download_filename = "suppliers-on-{}.csv".format(framework_slug)
+
+    formatted_rows = []
+    for row in supplier_rows:
+        formatted_rows.append(
+            [row[heading] for heading in supplier_and_framework_headers] +
+            [row['published_services_count'][heading] for heading in service_count_headers] +
+            [row['contact_information'][heading] for heading in contact_info_headers]
+        )
+    formatted_rows.insert(
```

Comment:
Reads a bit out of the logical order here. We're building the csv into the `formatted_rows` variable and then going back and inserting the header afterpopulating the rows. Even though the header doesn't depend on the rows.
It's a bit like building the csv backwards, also that way we need to use the `[].insert` method instead of keeping it the same as above and using `[].append`.

Reply:
Happy to change - as you can probably tell this was copied wholesale from the users export view above.